src/sql/src/plan/expr.rs

```diff
@@ -1386,6 +1523,584 @@ impl HirRelationExpr {
     }
 }
 
+impl VisitChildren<Self> for HirRelationExpr {
+    fn visit_children<F>(&self, mut f: F)
+    where
+        F: FnMut(&Self),
+    {
+        // subqueries of type HirRelationExpr might be wrapped in
+        // Exists or Select variants within HirScalarExpr trees
+        // attached at the current node, and we want to visit them as well
+        VisitChildren::visit_children(self, |expr: &HirScalarExpr| {
```

Comment:
Since `visit_{mut}_children` are now recursive, should we change its return type to `Result<(), RecursionLimitError>`, and introduce deprecated `_nolimit` versions, as we previously did for the non-fallible `Visit` methods?